SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/DynamicSubobjectsTest/DynamicSubobjectsTest.cpp

```diff
@@ -52,198 +70,251 @@ void ADynamicSubobjectsTest::PrepareTest()
 	const int DynamicComponentsPerClass = GetDefault<USpatialGDKSettings>()->MaxDynamicallyAttachedSubobjectsPerClass;
 	StepTimer = 0.0f;
 
-	{ // Step 0 - The server spawn a TestMovementCharacter and makes Client 1 possess it.
-		AddStep(TEXT("DynamicSubobjectsTestSetup"), FWorkerDefinition::Server(1), nullptr, [this]() {
-			ASpatialFunctionalTestFlowController* ClientOneFlowController = GetFlowController(ESpatialFunctionalTestWorkerType::Client, 1);
-			APlayerController* PlayerController = Cast<APlayerController>(ClientOneFlowController->GetOwner());
+	// Step 0 - The server spawn a TestMovementCharacter and makes Client 1 possess it.
+	AddStep(TEXT("DynamicSubobjectsTestSetup"), FWorkerDefinition::Server(1), nullptr, [this]() {
+		ASpatialFunctionalTestFlowController* ClientOneFlowController = GetFlowController(ESpatialFunctionalTestWorkerType::Client, 1);
+		APlayerController* PlayerController = Cast<APlayerController>(ClientOneFlowController->GetOwner());
 
-			if (IsValid(PlayerController))
-			{
-				ClientOneSpawnedPawn =
-					GetWorld()->SpawnActor<ATestMovementCharacter>(CharacterSpawnLocation, FRotator::ZeroRotator, FActorSpawnParameters());
-				RegisterAutoDestroyActor(ClientOneSpawnedPawn);
+		if (AssertIsValid(PlayerController, TEXT("PlayerController should be valid")))
+		{
+			ClientOneSpawnedPawn = GetWorld()->SpawnActor<ATestMovementCharacter>(CharacterSpawnLocation, FRotator::ZeroRotator);
+			RegisterAutoDestroyActor(ClientOneSpawnedPawn);
 
-				ClientOneDefaultPawn = PlayerController->GetPawn();
+			ClientOneDefaultPawn = PlayerController->GetPawn();
+			PlayerController->Possess(ClientOneSpawnedPawn);
 
-				PlayerController->Possess(ClientOneSpawnedPawn);
+			FinishStep();
+		}
+	});
+
+	// Step 1 - All workers check if they have one AReplicatedGASTestActor in the world, and set a reference to it.
+	AddStep(
+		TEXT("DynamicSubobjectsTestAllWorkers"), FWorkerDefinition::AllWorkers, nullptr, nullptr,
+		[this](float DeltaTime) {
+			TestActor = GetReplicatedTestActor();
+			FinishStep();
+		},
+		StepTimeLimit);
 
+	// Step 2 - Client 1 checks if it has correctly possessed the TestMovementCharacter.
+	AddStep(
+		TEXT("DynamicSubobjectsTestClientCheckPossesion"), FWorkerDefinition::Client(1), nullptr, nullptr,
+		[this](float DeltaTime) {
+			APawn* PlayerCharacter = GetFlowPawn();
+			if (AssertIsValid(PlayerCharacter, TEXT("PlayerCharacter should be valid"))
```

Comment:
Since we only assert in this step, it can be a non-ticking step (second argument instead of third). Can also remove time limit in that case. Asserts will fail the build regardless, so don't have to do an if.
Can just be
````
assert()
finishstep()
```

Reply:
ah, nvm, missed that we also check the pawn
should be converted to another assert, I think